seisbench/models/base.py

```diff
@@ -192,6 +193,113 @@ def download_metadata_callback(metadata_path):
 
         return weights
 
+    @classmethod
+    def load(cls, path):
+        """
+        Load a SeisBench model from local path.
+
+        For more information on the SeisBench model format see py:func:`save`.
+
+        :param path: Define the path to the SeisBench model directory.
+        :type path: pathlib.Path
+        :return: Model instance
+        :rtype: SeisBenchModel
+        """
+        model_files = os.listdir(path)
+
+        if sum([file.endswith("pt") for file in model_files]) > 1:
+            seisbench.logger.warning(
+                f"Strange format of SeisBench model directory detected at {path}. "
+                f"There is more than one .pt file. This will lead to unforseen "
+                f"behaviour during model loading."
+            )
+        if sum([file.endswith("json") for file in model_files]) > 1:
+            seisbench.logger.warning(
+                f"Strange format of SeisBench model directory detected at {path}. "
+                f"There is more than one .json file. This will lead to unforseen "
+                f"behaviour during model loading."
+            )
+
+        for file in model_files:
+            if file.endswith("json"):
+                with open(path / file, "r") as json_file:
+                    weights_metadata = json.load(json_file)
+            elif file.endswith("pt"):
+                model_weights = torch.load(f"{path / file}")
+
+        def _get_args(obj):
+            signature = inspect.signature(obj)
+            args = {k: v._default for k, v in signature.parameters.items()}
+            if "kwargs" in args.keys():
+                del args["kwargs"]
+            return args
+
+        model_args = weights_metadata.get("model_args", {})
+
+        # Overwrite default class args from config model args
```

Comment:
I'm not really sure why this part is needed. Could you explain this?

Reply:
Ok, so this comes about because when we save out the model we do this by tracking all the state during the ```WaveformModel``` constructor call with ```self._argspec = locals()```. 

This means there are extra configuration parameters in the ```self._argspec``` dict which belong to the parent class ```WaveformModel```. We do this here as we need to provide a layer of abstraction so that all inherited models do not have to implement the ```locals()``` call themselves. 

We then just 'cherry-pick' or extract out the arguments needed to pass to the model constructor only.

Reply:
After reading your solution to this problem of identifying what arguments are relevant for saving below, the above code is just my current approach to do this. With the outlined drawback of not being able to detect parameters changed after model construction a current limitation during saving.